src/ansys/fluent/core/streaming_services/events_streaming.py

```diff
@@ -225,3 +243,64 @@ def start(self, *args, **kwargs) -> None:
     def stop(self) -> None:
         """Stop streaming."""
         self._impl.stop()
+
+    def _register_solution_event_sync_callback(
+        self,
+        event_type: Literal[
+            SolverEvent.ITERATION_ENDED_SYNC, SolverEvent.TIMESTEP_ENDED_SYNC
+        ],
+        callback_id: str,
+        callback: Callable,
+    ) -> tuple[Literal[SolverEvent.SOLUTION_PAUSED], Callable]:
+        unique_id = self._session.scheme_eval.scheme_eval(
+            f"""
+            (let
+                ((ids
+                    (let loop ((i 1))
+                        (define next-id (string->symbol (format #f "pyfluent-~d" i)))
+                        (if (check-monitor-existence next-id)
+                            (loop (1+ i))
+                            (list i next-id)
+                            )
+                        )
+                    ))
+                (register-solution-monitor
+                    (cadr ids)
+                    (lambda (niter time)
+                        (if (integer? niter)
+                            (begin
+                                (events/transmit 'auto-pause (cons (car ids) niter))
+                                (grpcserver/auto-pause (is-server-running?) (cadr ids))
+                                )
+                            )
+                        ()
+                        )
+                    {'#t' if event_type == SolverEvent.TIMESTEP_ENDED_SYNC else '#f'}
+                    )
+                (car ids)
+                )
+        """
+        )
```

Comment:
😮

Reply:
An alternative simpler strategy could be to send `uuid1` from clients, but the size of `uuid1` won't fit in the auto-pause response.